lib/vector-core/src/config/proxy.rs

```diff
@@ -45,6 +45,7 @@ impl NoProxyInterceptor {
 /// to use based on the type of traffic being proxied, as well as set specific hosts that
 /// should not be proxied.
 #[configurable_component]
+#[configurable(metadata(docs::advanced))]
```

Comment:
what does/will that do?

Reply:
We're using it to better control the config builder UI, akin to the "common" attribute the vector docs have now. To be used for "most of the time you don't need this" so the UX is better.

Reply:
This. We'll eventually make the documentation generation script start using it, but this is laying the groundwork.